src/analysis/normed_space/normed_group_hom.lean

```diff
@@ -0,0 +1,450 @@
+  mul_le_mul_of_nonneg_right (le_max_left _ _) (norm_nonneg x)
+
+alias mk_continuous_norm_le ← add_monoid_hom.mk_continuous_norm_le
+alias mk_continuous_norm_le' ← add_monoid_hom.mk_continuous_norm_le'
+
+/-! ### Addition of normed group homs -/
+
+/-- Addition of normed group homs. -/
+instance : has_add (normed_group_hom V₁ V₂) :=
+⟨λ f g, (f.to_add_monoid_hom + g.to_add_monoid_hom).mk_continuous (∥f∥ + ∥g∥) $ λ v, calc
+  ∥f v + g v∥
+      ≤ ∥f v∥ + ∥g v∥ : norm_add_le _ _
+  ... ≤ ∥f∥ * ∥v∥ + ∥g∥ * ∥v∥ : add_le_add (le_op_norm f v) (le_op_norm g v)
+  ... = (∥f∥ + ∥g∥) * ∥v∥ : by rw add_mul⟩
+
+/-- The operator norm satisfies the triangle inequality. -/
+theorem op_norm_add_le : ∥f + g∥ ≤ ∥f∥ + ∥g∥ :=
+mk_continuous_norm_le _ (add_nonneg (op_norm_nonneg _) (op_norm_nonneg _)) _
+
+@[simp] lemma coe_add (f g : normed_group_hom V₁ V₂) : ⇑(f + g) = f + g := rfl
+lemma add_apply (f g : normed_group_hom V₁ V₂) (v : V₁) :
+  ((f + g : normed_group_hom V₁ V₂)) v = f v + g v := rfl
+
+/-! ### The zero normed group hom -/
+
+instance : has_zero (normed_group_hom V₁ V₂) :=
+⟨(0 : V₁ →+ V₂).mk_continuous 0 (by simp)⟩
+
+instance : inhabited (normed_group_hom V₁ V₂) := ⟨0⟩
+
+/-- An operator is zero iff its norm vanishes. -/
+theorem op_norm_zero_iff : ∥f∥ = 0 ↔ f = 0 :=
+iff.intro
+  (λ hn, ext (λ x, norm_le_zero_iff.1
+    (calc _ ≤ ∥f∥ * ∥x∥ : le_op_norm _ _
+     ...     = _ : by rw [hn, zero_mul])))
+  (λ hf, le_antisymm (real.Inf_le _ bounds_bdd_below
+    ⟨ge_of_eq rfl, λ _, le_of_eq (by { rw [zero_mul, hf], exact norm_zero })⟩)
+    (op_norm_nonneg _))
+
+@[simp] lemma coe_zero : ⇑(0 : normed_group_hom V₁ V₂) = 0 := rfl
+lemma zero_apply (v : V₁) : ((0 : normed_group_hom V₁ V₂)) v = 0 := rfl
+
+variables {f g}
+
+/-! ### The identity normed group hom -/
+
+/-- The identity as a continuous normed group hom. -/
+@[simps]
+def id : normed_group_hom V V :=
+(add_monoid_hom.id V).mk_continuous 1 (by simp [le_refl])
+
+/-- The norm of the identity is at most `1`. It is in fact `1`, except when the space is trivial
+where it is `0`. It means that one can not do better than an inequality in general. -/
+lemma norm_id_le : ∥(id : normed_group_hom V V)∥ ≤ 1 :=
+op_norm_le_bound _ zero_le_one (λx, by simp)
+
+/-- If a space is non-trivial, then the norm of the identity equals `1`. -/
+lemma norm_id [nontrivial V] : ∥(id : normed_group_hom V V)∥ = 1 :=
+le_antisymm norm_id_le $ let ⟨x, hx⟩ := exists_ne (0 : V) in
+have _ := (id : normed_group_hom V V).ratio_le_op_norm x,
+by rwa [id_apply, div_self (ne_of_gt $ norm_pos_iff.2 hx)] at this
+
+/-! ### The negation of a normed group hom -/
+
+/-- Opposite of a normed group hom. -/
+instance : has_neg (normed_group_hom V₁ V₂) :=
+⟨λ f, (-f.to_add_monoid_hom).mk_continuous (∥f∥) (λ v, by simp [le_op_norm f v])⟩
+
+@[simp] lemma coe_neg (f : normed_group_hom V₁ V₂) : ⇑(-f) = -f := rfl
+lemma neg_apply (f : normed_group_hom V₁ V₂) (v : V₁) :
+  ((-f : normed_group_hom V₁ V₂)) v = - (f v) := rfl
```

Comment:
```suggestion
  (-f : normed_group_hom V₁ V₂) v = - (f v) := rfl
```
although I think this might work?
```suggestion
  (-f : _) v = - (f v) := rfl
```

Reply:
I think it doesn't hurt to use the explicit version.

Reply:
There are some double-parentheses in the other lemmas too

Reply:
I removed the double parentheses